File: src/com/leetcode/Medium/SearchA2DMatrix.java
package com.leetcode.Medium;

// 74. Search a 2D Matrix
public class SearchA2DMatrix {
    public boolean searchMatrix(int[][] matrix, int target) {
        for (int[] ints : matrix) {
            int low = 0, high = ints.length - 1;
            while (low <= high) {
                int mid = (low + high) / 2;
                if (ints[mid] == target) {
                    return true;
                } else if (target < ints[mid]) {
                    high = mid - 1;
                } else {
                    low = mid + 1;
                }
            }
        }
        return false;
    }
}

File: JavaScript/Medium/SearchA2DMatrix.js
/**
 * 74. Search a 2D Matrix
 * @param {number[][]} matrix
 * @param {number} target
 * @return {boolean}
 */
const searchMatrix = function (matrix, target) {
  for (let i = 0; i < matrix.length; i++) {
    let low = 0, high = matrix[i].length - 1;
    while (low <= high) {
      let mid = Math.floor((low + high) / 2);
      if (matrix[i][mid] == target) {
        return true;
      } else if (target < matrix[i][mid]) {
        high = mid - 1;
      } else {
        low = mid + 1;
      }
    }
  }
  return false;
};
